search-app/src/cmr/search/results_handlers/csv_results_handler.clj

```diff
@@ -1,16 +1,58 @@
 (ns cmr.search.results-handlers.csv-results-handler
   "Handles the CSV results format and related functions."
-  (:require [cmr.common-app.services.search.elastic-results-to-query-results :as elastic-results]
-            [cmr.common-app.services.search.elastic-search-index :as elastic-search-index]
-            [cmr.common-app.services.search :as qs]
-            [cheshire.core :as json]
-            [clojure.data.csv :as csv]
-            [clojure.string :as str]
-            [cmr.search.services.acls.acl-results-handler-helper :as acl-rhh])
+  (:require
+   [cheshire.core :as json]
+   [clojure.data.csv :as csv]
+   [camel-snake-kebab.core :as csk]
+   [clojure.string :as str]
+   [cmr.common-app.services.search :as qs]
+   [cmr.common-app.services.search.elastic-results-to-query-results :as elastic-results]
+   [cmr.common-app.services.search.elastic-search-index :as elastic-search-index]
+   [cmr.search.services.acls.acl-results-handler-helper :as acl-rhh])
   (:import
     [java.io StringWriter]))
 
-(def CSV_HEADER
+(def COLLECTION_CSV_HEADER
+  ["Data Provider"
+   "Short Name"
+   "Version"
+   "Entry Title"
+   "Processing Level"
+   "Platform"
+   "Start Time"])
+
+(defmethod elastic-search-index/concept-type+result-format->fields [:collection :csv]
+ [concept-type query]
+ (->> COLLECTION_CSV_HEADER
+      (remove #(= "Data Provider" %))
+      (map csk/->kebab-case)
+      (concat acl-rhh/collection-elastic-fields)
+      distinct))
+
+(defmethod elastic-results/elastic-result->query-result-item [:collection :csv]
+  [context query elastic-result]
+  (let [{start-date :start-date
+         provider-id :provider-id
+         processing-level :processing-level-id
+         entry-title :entry-title
+         platforms :platform
+         version :version-id
+         short-name :collection-short-name} (:_source elastic-result)
+        start-date (when start-date (str/replace (str start-date) #"\+0000" "Z"))]
+    (merge {:row [provider-id short-name version entry-title processing-level (first platforms) start-date]}
```

Comment:
`platforms` are to my knowledge the only thing coming back from a CSV result set that are a sequence. Should I take the first item like I already am, or combine the values somehow? We're just expecting the short name

Reply:
We should return the platforms as comma separated string when there are multiple. Refer to granule `Online Access URLs` and `Browse URLs` for examples.